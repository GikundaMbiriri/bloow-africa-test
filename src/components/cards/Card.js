import React , { useEffect } from 'react'
import AudioPlayer from 'react-h5-audio-player';
import 'react-h5-audio-player/lib/styles.css';
import { HeartIcon,ShareIcon } from '@heroicons/react/solid'
import { HeartIcon as Heart } from '@heroicons/react/outline'
import './card.css'
import "react-loader-spinner/dist/loader/css/react-spinner-loader.css";
import { Audio } from  'react-loader-spinner'
import { getSongs,likeSong,unlikeSong } from "../../redux/actions/dataAction";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import About from '../about/About'
import { Link } from 'react-router-dom';
function Card(props) {
  useEffect(() => {
    props.getSongs();
  }, []);
  const { songs, loading } = props.data;
  const { authenticated, likes } = props.user;

const likedSong=(song)=>{
  if(likes&& likes.find(like=>like.songId===song.songId)){
   
    return true;

  }
  else {
    
   

    return false;}

  
}
const likeSong=(song)=>{
  props.likeSong(song.songId)
}
const unlikeSong=(song)=>{
  props.unlikeSong(song.songId)
}
  return (
    <>
    <div className='md:w-2/5 pr-5 pt-5 m-auto w-11/12 overflow-x-hidden'>
        <h1 className='text-2xl font-bold pb-3 text-center text-yellow-600'>Cool Songs</h1>
        {!!songs[0]?songs.map((song)=>
  <AudioPlayer
  className=' h-55 my-5'
  
   
    header={<div className='flex justify-between'><div className='w-1/2'>
      <h2 className='font-semibold  text-xl text-dark-blue'>{song.songName}<span className='text-dark-blue'> by</span>  <span className='text-lg text-dark-blue italic'>{song.artist}</span> </h2>
      </div><div className='flex justify-around items-center w-3/10 px-2'>
      {!authenticated?(<Link to='/login'><div data-testid='borderedHeart'><Heart className="h-6 w-6 text-black cursor-pointer" /></div></Link>):
        (likedSong(song)?
        (<HeartIcon className="h-6 w-6 text-dark-blue cursor-pointer" data-testid='filledHeart' onClick={()=>unlikeSong(song)}/>):
        (<Heart className="h-6 w-6 text-black cursor-pointer" onClick={()=>likeSong(song)}/>))} <div>{song.likeCount}</div> 
       </div></div>}
 key={song.songId}
    src={song.songUrl}
    onPlay={e => console.log("onPlay")}
   
  />
        ):songs.length==0?<div datat-testid='emptySongs'>No songs found</div>:
        <div  className='h-55 w-100 flex justify-between' data-testid='loading'>
        <Audio
      className='w-50'
       color='black'
        ariaLabel='loading'
       
      />
    </div>
      }


  <div  className='block md:hidden'>
  <About />
  </div>
  </div>
    </>
  )
}


Card.propTypes = {
  getSongs: PropTypes.func.isRequired,
  likeSong: PropTypes.func.isRequired,
  unlikeSong: PropTypes.func.isRequired,
  data: PropTypes.object.isRequired,
  user:PropTypes.object.isRequired,
};
const mapStateToProps = (state) => ({
  data: state.data,
  user:state.user
});
const mapActionsToProps={
  getSongs,
  likeSong,
  unlikeSong
}
export default connect(mapStateToProps, mapActionsToProps)(Card);
