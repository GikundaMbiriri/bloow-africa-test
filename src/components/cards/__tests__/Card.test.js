import React from 'react'
import ReactDOM from 'react-dom'
import Card from '../Card';
import {render as rtlRender, screen} from '@testing-library/react'
import renderer from 'react-test-renderer';
import store from '../../../redux/store';
import {Provider} from 'react-redux';
import { HashRouter as Router } from "react-router-dom";
import { act } from '@testing-library/react-hooks/dom'
const render=component=>rtlRender(
    <Provider store={store}>
        <Router>
        {component}
        </Router>
    </Provider>
)
it("it renders without crashing",()=>{
    const div = document.createElement('div');
    render(<Card></Card>,div)
})

it('renders correctly', () => {
    const tree = renderer
      .create(
      <Provider store={store}>
          <Router>
      <Card></Card>
      </Router>
      </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });


