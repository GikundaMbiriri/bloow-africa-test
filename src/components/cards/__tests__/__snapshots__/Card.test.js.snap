// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="md:w-2/5 pr-5 pt-5 mx-auto w-11/12 overflow-x-hidden"
>
  <h1
    className="text-2xl font-bold pb-3  text-center text-yellow-600"
  >
    Cool Songs
  </h1>
  <div
    className="h-100 w-100 flex justify-around items-center"
    data-testid="loading"
  >
    <div
      data-testid="audio-loading"
      style={
        Object {
          "display": "flex",
        }
      }
    >
      <svg
        aria-label="loading"
        data-testid="audio-svg"
        fill="black"
        height="100"
        viewBox="0 0 55 80"
        width="100"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          transform="matrix(1 0 0 -1 0 80)"
        >
          <rect
            height="20"
            rx="3"
            width="10"
          >
            <animate
              attributeName="height"
              begin="0s"
              calcMode="linear"
              dur="4.3s"
              repeatCount="indefinite"
              values="20;45;57;80;64;32;66;45;64;23;66;13;64;56;34;34;2;23;76;79;20"
            />
          </rect>
          <rect
            height="80"
            rx="3"
            width="10"
            x="15"
          >
            <animate
              attributeName="height"
              begin="0s"
              calcMode="linear"
              dur="2s"
              repeatCount="indefinite"
              values="80;55;33;5;75;23;73;33;12;14;60;80"
            />
          </rect>
          <rect
            height="50"
            rx="3"
            width="10"
            x="30"
          >
            <animate
              attributeName="height"
              begin="0s"
              calcMode="linear"
              dur="1.4s"
              repeatCount="indefinite"
              values="50;34;78;23;56;23;34;76;80;54;21;50"
            />
          </rect>
          <rect
            height="30"
            rx="3"
            width="10"
            x="45"
          >
            <animate
              attributeName="height"
              begin="0s"
              calcMode="linear"
              dur="2s"
              repeatCount="indefinite"
              values="30;45;13;80;56;72;45;76;34;23;67;30"
            />
          </rect>
        </g>
      </svg>
    </div>
  </div>
  <div
    className="block md:hidden"
  >
    <div
      className="w-full"
    >
      <h1
        className="text-2xl font-bold py-3 text-center text-dark-blue"
      >
        About Bloow Africa
      </h1>
      <ul
        className="list-disc w-2/3 m-auto"
      >
        <li
          className="pb-2"
        >
          We  aim at developing the best music platform for artists world wide and locally.
        </li>
        <li
          className="pb-2"
        >
          We  aim at developing the best music platform for artists world wide and locally.
        </li>
        <li
          className="pb-2"
        >
          We  aim at developing the best music platform for artists world wide and locally.
        </li>
        <li
          className="pb-2"
        >
          We  aim at developing the best music platform for artists world wide and locally.
        </li>
      </ul>
    </div>
  </div>
</div>
`;
